### DISCUSSION TITLE
[QUESTION] A configuration which can make volume/replica meta management in the same node

### DISCUSSION THREAD

**Discussion by zhaohongda1987** [longhorn/longhorn#6807] [Q&A]
## Environment

 - Longhorn version: 1.51
 - Kubernetes version: 1.27
 - Node config
   - OS type and version
   - Kernel version
   - CPU per node:
   - Memory per node:
   - Disk type
   - Network bandwidth and latency between the nodes:
 - Underlying Infrastructure (e.g. on AWS/GCE, EKS/GKE, VMWare/KVM, Baremetal): AWS

## Additional context
For longhorn architecture as below picture, I know that the replica management may in another node, because its a distributed System. 
![image](https://github.com/longhorn/longhorn/assets/19568031/b9ec6853-503e-4b35-a3b7-032ae496632b)
But we suggest to add a configuration which like data locality settings, it can force make the manager just maintain the replica which are in the same node. So that the file metadata transfer/receive not depend on network, but just run inside the node. would it possible? thanks
Scenarios for us: use SSD disk for cache so that we want IOPS/Bandwidth performance lost as less as possible

**Comment by zhaohongda1987** [comment]
For now, we set data locality and replica just set to 1, that means k8s pod and pvc in the same node with 1 replica. We test the performance and found that the IOPS/Bandwidth lost may much more by more small files, because the file metadata need to transfer to another node's instance-manager

**Reply by c3y1huang** [reply]
Have you tried [Local Volume via Data Locality Strict Mode](https://longhorn.io/blog/longhorn-v1.4/#local-volume-via-data-locality-strict-mode), which was introduced in v.1.4?

https://longhorn.io/docs/1.5.1/references/settings/#default-data-locality
